src/analysis/normed_space/ordered.lean

```diff
@@ -48,3 +67,146 @@ instance : normed_linear_ordered_field ℚ :=
+begin
+  rw le_antisymm_iff,
+  split,
+  { apply normed_lattice_add_comm_group.solid,
+    rw ← lattice_ordered_comm_group.abs_idempotent a, },
+  { apply normed_lattice_add_comm_group.solid,
+    rw ← lattice_ordered_comm_group.abs_idempotent a, }
+end
+
+/--
+Let `α` be a normed lattice ordered group. Then the infimum is jointly continuous.
+-/
+@[priority 100] -- see Note [lower instance priority]
+instance normed_lattice_add_comm_group_has_continuous_inf {α : Type*}
+  [normed_lattice_add_comm_group α] : has_continuous_inf α :=
+⟨ continuous_iff_continuous_at.2 $ λ q, tendsto_iff_norm_tendsto_zero.2 $
+begin
+  have : ∀ p : α × α, ∥p.1 ⊓ p.2 - q.1 ⊓ q.2∥ ≤ ∥p.1 - q.1∥ + ∥p.2 - q.2∥,
+  {
+    intros,
+    nth_rewrite_rhs 0  ← norm_abs_eq_norm,
+    nth_rewrite_rhs 1  ← norm_abs_eq_norm,
+    apply le_trans _ (norm_add_le (|p.fst - q.fst|) (|p.snd - q.snd|)),
+    apply normed_lattice_add_comm_group.solid,
+    rw lattice_ordered_comm_group.abs_pos_eq (|p.fst - q.fst| + |p.snd - q.snd|),
+    { calc |p.fst ⊓ p.snd - q.fst ⊓ q.snd| =
+        |p.fst ⊓ p.snd - q.fst ⊓ p.snd + (q.fst ⊓ p.snd - q.fst ⊓ q.snd)| :
+          by { rw sub_add_sub_cancel, }
+        ... ≤ |p.fst ⊓ p.snd - q.fst ⊓ p.snd| + |q.fst ⊓ p.snd - q.fst ⊓ q.snd| :
+          by {apply lattice_ordered_comm_group.abs_triangle,}
+        ... ≤ |p.fst - q.fst | + |p.snd - q.snd| : by {
+          apply add_le_add,
+          apply
+            (sup_le_iff.elim_left (lattice_ordered_comm_group.Birkhoff_inequalities _ _ _)).right,
+          -- Should I be surprised that this isn't infered automatically?
+          apply normed_lattice_add_comm_group_to_covariant_class,
+          rw inf_comm,
+          nth_rewrite 1 inf_comm,
+          apply
+            (sup_le_iff.elim_left (lattice_ordered_comm_group.Birkhoff_inequalities _ _ _)).right,
+          -- Should I be surprised that this isn't infered automatically?
+          apply normed_lattice_add_comm_group_to_covariant_class,
+        }, },
+    { apply add_nonneg,
+      apply lattice_ordered_comm_group.abs_pos,
+      apply lattice_ordered_comm_group.abs_pos, }
+  },
+  refine squeeze_zero (λ e, norm_nonneg _) this _,
+  convert (((continuous_fst.tendsto q).sub tendsto_const_nhds).norm).add
+        (((continuous_snd.tendsto q).sub tendsto_const_nhds).norm),
+  simp,
+end
+⟩
+
+/--
+Let `α` be a normed lattice ordered group. Then the supremum is jointly continuous.
+-/
+lemma normed_lattice_add_comm_group_has_continuous_sup {α : Type*}
+  [h: normed_lattice_add_comm_group α] : has_continuous_sup α :=
+infer_instance
+
+/--
+Let `α` be a normed lattice ordered group. Then `α` is a topological lattice in the norm topology.
+-/
+@[priority 100] -- see Note [lower instance priority]
+instance normed_lattice_add_comm_group_topological_lattice {α : Type*}
+[h: normed_lattice_add_comm_group α] : topological_lattice α := by exact topological_lattice.mk
```

Comment:
```suggestion
  [h: normed_lattice_add_comm_group α] : topological_lattice α := 
by exact topological_lattice.mk
```

Reply:
Also, no need for `by exact ` here: you can just write the term.

Reply:
Done.